File: Ejercicio 003/damian.php
<?php
//Resolucion del ejercicio utilizando recursividad
function countBits($n) {
    if($n==0) {
        return 0;
    }else{
        if($n%2!=0){ // Si el resto de  dividir por dos es distinto de cero (n es impar), esto significa que el bit menos significativo es igual a 1
        return 1 + countBits($n/2);
        }else{ //El numero es par (bit menos significativo igual a cero)
        return countBits($n/2);    
        } 
    }
}

    

File: Ejercicio 003/damian.c
//Resolucion del ejercicio operando a nivel de bits
#include <stddef.h> //incluye la libreria

size_t countBits(unsigned value){
  int bits_uno =0; //inicializamos la variable para contar unos
  while(value | 0 != 0){ //La condicion evalúa si el valor binario de value esta compuesto solo por ceros
    if (value & 1 == 1){  //La condicion evalúa si el bit menos significatio de value es igual a uno
      bits_uno ++;  //incrementa en uno nuestro contador
    }
    value = value>>1; //Desplazamos el cursor un valor a la derecha (por ej pasa de 0111 a 0011; de 0100 a 0010, etc)
  }
	return bits_uno;
}